modules/services/nix-daemon.nix

```diff
@@ -51,7 +51,7 @@ in
       serviceConfig.ProcessType = config.nix.daemonProcessType;
       serviceConfig.LowPriorityIO = config.nix.daemonIOLowPriority;
       serviceConfig.Label = "org.nixos.nix-daemon"; # must match daemon installed by Nix regardless of the launchd label Prefix
-      serviceConfig.SoftResourceLimits.NumberOfFiles = mkDefault 4096;
+      serviceConfig.SoftResourceLimits.NumberOfFiles = mkDefault 1048576;
```

Comment:
In [nixpkgs stdenv room](https://matrix.to/#/!OqhvaDMJdKYUicLDiE:nixos.org/$wdXfRpNPeyPtG1BsnTeMXVD7YcAHRqjolcWN1gR5YnA?via=nixos.org&via=matrix.org&via=nixos.dev) we're [wondering if this is even doing anything](https://matrix.to/#/!OqhvaDMJdKYUicLDiE:nixos.org/$asZIEr-iAzHfBA3OTjj3dYNsy_ZSLHuwA_cK58d7cso?via=nixos.org&via=matrix.org&via=nixos.dev)
because it's above `kern.maxfilesperproc` on various darwins and [might be getting ignored](https://matrix.to/#/!OqhvaDMJdKYUicLDiE:nixos.org/$Yd8xzcj6us7lSwcalzc5qZFjnjDUShW4tK_Fn9EyuaY?via=nixos.org&via=matrix.org&via=nixos.dev) or something?